src/Core/src/Platform/Android/ViewExtensions.cs

```diff
@@ -198,17 +197,39 @@ internal static void UpdateBackground(this EditText platformView, IView view)
 				mauiDrawable.Dispose();
 			}
 
-			// Android will reset the padding when setting a Background drawable
+			if (platformView.Background is LayerDrawable layerDrawable)
+			{
+				platformView.Background = null;
+				layerDrawable.Dispose();
+			}
+
+			// Android will reset the padding when setting a Background drawable	
 			// So we need to reapply the padding after
 			var padLeft = platformView.PaddingLeft;
 			var padTop = platformView.PaddingTop;
 			var padRight = platformView.PaddingRight;
 			var padBottom = platformView.PaddingBottom;
 
-			var previousDrawable = platformView.Background;
+			var paint = view.Background;
+
+			Drawable? defaultBackgroundDrawable = ContextCompat.GetDrawable(platformView.Context, Resource.Drawable.abc_edit_text_material);
+
+			var previousDrawable = defaultBackgroundDrawable ?? platformView.Background;
 			var backgroundDrawable = paint!.ToDrawable(platformView.Context);
-			LayerDrawable layer = new LayerDrawable(new Drawable[] { backgroundDrawable!, previousDrawable! });
-			platformView.Background = layer;
+
+			if (previousDrawable is null)
+				platformView.Background = backgroundDrawable;
+			else
+			{
+				if (paint.IsNullOrEmpty())
+					platformView.Background = previousDrawable;
+				else
+				{
+
+					LayerDrawable layer = new LayerDrawable(new Drawable[] { backgroundDrawable!, previousDrawable });
```

Comment:
Can we remove all the null forgiveness operators from this code?

line 218 as well?

Reply:
![61](https://user-images.githubusercontent.com/6755973/228222950-9295752d-2e08-4b83-b335-5471c38a2dc6.png)

Reply:
The `ToDrawable` method extension return a `Drawable?`. In that case we create a `LayerDrawable` setting the default previous Drawable (the default background with the bottom line).